src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -193,6 +220,52 @@ snc_quota_manager::get_quota_balancer_node_period() const {
     return v;
 }
 
+void snc_quota_manager::update_node_quota_default() {
+    const ingress_egress_state<std::optional<quota_t>> new_node_quota_default
+      = calc_node_quota_default();
+    if (ss::this_shard_id() == quota_balancer_shard) {
+        // downstream updates:
+        // - shard effective quota (via _shard_quotas_update):
+        const auto calc_diff = [](
+                                 const std::optional<quota_t>& qold,
+                                 const std::optional<quota_t>& qnew) {
+            dispense_quota_amount diff;
+            if (qold && qnew) {
+                diff.amount = *qnew - *qold;
+                diff.amount_is = dispense_quota_amount::amount_t::delta;
+            } else if (qold || qnew) {
+                diff.amount = node_to_shard_quota(qnew);
+                diff.amount_is = dispense_quota_amount::amount_t::value;
+            }
+            return diff;
+        };
+        ingress_egress_state<dispense_quota_amount> shard_quotas_update{
+          .in = calc_diff(_node_quota_default.in, new_node_quota_default.in),
+          .eg = calc_diff(_node_quota_default.eg, new_node_quota_default.eg)};
+        ssx::spawn_with_gate(
+          _balancer_gate, [this, update = shard_quotas_update] {
+              return quota_balancer_update(update);
```

Comment:
Is another background fiber necessary? Its not like it can increase the processing throughput rate, seems like its just for when config updates arrive. These can be processed by the main fiber.

Reply:
That's right they are for config updates. Since the config updates include as many cross core calls as a balancer step does -- if not more -- it was my assumption there is no need to make the property change notifications wait for that to complete.

However there is a stronger argument for asynchronity here: a balancer step and a config update step should not run concurently, and I suspect we definitely won't appreciate a mutex in the main fiber. The concurrency may lead to various inconsistencies in quota sharing between shards. The problems root from the fact that updates to quotas can be both assignments and increments (or decrements). When assignment and increments are applied concurrently by separate fibers, the result is undefined.

Reply:
>a balancer step and a config update step should not run concurrently

agree, but this presumes a design in which these steps both launch background fibers, which then need to be synchronized. i believe a design exists in which a configuration update merely triggers an iteration of the reconciliation loop (e.g. an early execution of the balancer step) which handles all cases.

in the end, i think this boils down to a design decision. imo @graphcareful  has a good point: if one fiber can correctly reconcile the state of the system with the state of the configuration, then there is no need for two fibers and thus no need for a mutex.

but it's also entirely valid to design this in such a way that the config change reaction and balancer steps are independent processes, which then need to be executed with mutual exclusion for correctness reasons. they are both valid solutions.